### PR TITLE
Update @wordpress/editor and remove @types/wordpress__editor

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Following the PR I created in Gutenberg ([Fix TS types for editor package](https://github.com/WordPress/gutenberg/pull/66754)), we no longer need buggy assertions in TS code.

## Proposed changes:

* Update `@wordpress/editor` to the latest version
* Remove `@types/wordpress__editor` which is no longer needed
* Fix the TS error following the update

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Just check if CI is happy

## What is next?
Following this PR we should replace the string selector (`'core/editor'`) with `store` from `@wordpress/editor` to ensure that the types work correctly. For example, in `projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/ai-image/featured-image.tsx`.

CC: @Automattic/jetpack-agora

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/ai-content-lens/extend/ai-post-excerpt/index.tsx [hunk 1]
```diff
@@ -308,14 +305,18 @@ export const PluginDocumentSettingPanelAiExcerpt = () => {
...
+				<PluginDocumentSettingPanel
+					className={ isBetaExtension( 'ai-content-lens' ) ? 'is-beta-extension inset-shadow' : '' }
+					name="ai-content-lens-plugin"
+					title={ __( 'Excerpt', 'jetpack' ) }
+				>
+					<AiPostExcerpt />
+				</PluginDocumentSettingPanel>
+			}
+		/>
 	);
```
Comment (manzoorwanijk) [inline projects/plugins/jetpack/extensions/plugins/ai-content-lens/extend/ai-post-excerpt/index.tsx:321]: There are many such issues in various components from `@wordpress/editor` package. I have created a PR in Gutenberg to fix those - https://github.com/WordPress/gutenberg/pull/67196

Comment (anomiex) [reply projects/plugins/jetpack/extensions/plugins/ai-content-lens/extend/ai-post-excerpt/index.tsx:321]: Is the switch here from `<Parent><Child/></Parent>` syntax to `<Parent children={ <Child /> } />` just to give it a place to add the `@ts-expect-error` comment, or is there a greater significance?

**Review by anomiex — APPROVED** [review]
Nice, glad to get rid of the `@types/` package. One more and we can lose [the hack for their outdated deps](https://github.com/Automattic/jetpack/blob/2bc0e23cf52c5949c0d403ba08aa307bd5151aea/.pnpmfile.cjs#L120-L129) too.

File: projects/plugins/jetpack/extensions/plugins/ai-content-lens/extend/ai-post-excerpt/index.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply projects/plugins/jetpack/extensions/plugins/ai-content-lens/extend/ai-post-excerpt/index.tsx:321]: > Is the switch here from `<Parent><Child/></Parent>` syntax to `<Parent children={ <Child /> } />` just to give it a place to add the `@ts-expect-error` comment

Yes, that is correct. There is no other reason.

**Pull request merged**